### PR TITLE
Uniswap connector changes

### LINKED ISSUE #31: New Uniswap connector
Take slippage in 2 ways for deposits.

Current version:-
tokenA has fixed amount but tokenB is variable according to price change in the meantime.

New version:- (Use try catch, no unitAmt needed only slippage)
- First try with tokenA fixed amount and tokenB variable. Let's take 1 ETH | 500 DAI with a slippage of 1%. So that makes the range at 1 ETH & 495 to 505 DAI. Now if ETH price increase meaning DAI between 500 to 505. The user might have a shortage of DAI in this scenario. If user has DAI shortage than next one.
- Take tokenB as fixed at 500 DAI & ETH will range from 0.99 to 1.01. So if above fails it'll succeed on this for sure.

### PR DESCRIPTION
**Pull request by thrilok209**
Closes #31 

- Added single token deposit function
- Updated deposit function logics

### CODE REVIEW

File: contracts/connectors/uniswap.sol [hunk 1]
```diff
@@ -144,6 +166,21 @@ contract UniswapHelpers is Stores, DSMath {
...
+        returns (uint256)
+    {
+         return
+            Babylonian
+                .sqrt(
+                    reserveIn.mul(
+                        userIn.mul(3988000).add(reserveIn.mul(3988009))
+                    )
+                ).sub(reserveIn.mul(1997)) / 1994;
+    }
```
Comment (KaymasJain) [inline contracts/connectors/uniswap.sol:190]: Can you explain what this function do?

File: contracts/connectors/uniswap.sol [hunk 2]
```diff
@@ -169,33 +206,119 @@ contract LiquidityHelpers is UniswapHelpers {
     function _addLiquidity(
         address tokenA,
         address tokenB,
-        uint _amt,
-        uint unitAmt,
-        uint slippage
+        uint amountADesired,
+        uint amountBDesired,
+        uint amountAMin,
+        uint amountBMin
```
Comment (KaymasJain) [inline contracts/connectors/uniswap.sol]: For easier calculation on the UI can we just have a slippage param? Like we'll send 1% from the UI. Will that work better?

Comment (thrilok209) [reply contracts/connectors/uniswap.sol]: Yeah. We can do that. I was also thinking the same.

File: contracts/connectors/uniswap.sol (hunk 1 above)
Comment (KaymasJain) [reply contracts/connectors/uniswap.sol:190]: @thrilok209?

Comment (thrilok209) [reply contracts/connectors/uniswap.sol:190]: This function will return how tokenA amount to swap to get tokenB amount in correct proportion with leftover tokenA amount after the swap.

**Pull request merged**